### ISSUE TITLE
VsVim StatusLine loads above horizontal scrollbar -- sometimes

### ISSUE THREAD

**Issue by gwojan** [jaredpar/VsVim#1659]
Most of the time after a project is loaded the VsVim statusline is located below the horizontal scrollbar. However, opening multiple VS sessions results in the statusline above the scrollbar. It seems totally inconsistent so I haven't been able to detect a pattern.

What is the correct location for the statusline?

I'm including a couple if images just for clarity.

![vsvim statusline below scrollbar](https://cloud.githubusercontent.com/assets/6855215/9306839/d914c18e-44c8-11e5-91b7-e06cf23ca69a.PNG)

![vsvim statusline above scrollbar](https://cloud.githubusercontent.com/assets/6855215/9306845/e355c2ba-44c8-11e5-83bc-128aa0a09e3a.PNG)

**Comment by jaredpar** [comment]
That's truly odd.  I've never seen that before and don't have a good idea when it would happen.  The placement of the command margin is declarative so it's odd that it would behave unpredictably in this manner.  Going to ask around to see if anyone has an idea of why this could be happening.

**Comment by jaredpar** [comment]
CC @jasonmalinowski in case he has an idea here.

**Comment by gwojan** [comment]
I guess I should ask which is the correct location declared for the statusline -- above or below the scrollbar? Which of the above images is "correct"?

**Comment by jasonmalinowski** [comment]
Margins are ordered the same way other orderable things are, with Name and Order attributes. If you didn't specify it, then you are probably at the mercy of the order MEF happens to import things. There's some description of that [here](https://msdn.microsoft.com/en-us/library/dd885244.aspx).

**Comment by jaredpar** [comment]
@jasonmalinowski do you know the name the scroll bar margin?  Or where I should go looking for it?

**Comment by jasonmalinowski** [comment]
Per the link I gave:

>These are the built-in margins:
>
>   * "Wpf Horizontal Scrollbar"
>   * "Wpf Vertical Scrollbar"
>   * "Wpf Line Number Margin"

Are these not working? Looks like there's also a [`PredefinedMarginNames`](https://msdn.microsoft.com/en-us/library/microsoft.visualstudio.text.editor.predefinedmarginnames.aspx) class.

**Comment by gwojan** [comment]
@jaredpar One thing I just noticed is if I don't have any additional windows like `Output`, `Error List` pinned like the images above show the VsVim margin consistently renders below the horizontal scroll bar.

If I pin something placement becomes random again.

**Comment by gwojan** [comment]
@jaredpar Ugh. Take my previous comment back... :frowning:

**Comment by jhamm** [comment]
I think this issue can be closed. The PR has been merged.

**Comment by jaredpar** [comment]
@jhamm i agree.  It seems really odd that this would have ever happened.  But the PR you submitted should close down the only case I could think of for causing this.

**Comment by gwojan** [comment]
@jaredpar, @jhamm, the problem still occurs on my system but I think it has something to do with the PowerShell Tools for Visual Studio. I loaded four different instances of Visual Studio 2013 including a PowerShell project and the StatusLine loaded below the horizontal scrollbar.

I closed all four instances and opened only a PowerShell project and the StatusLine appeared above the scrollbar. :frowning:
